### PR TITLE
Switch tests to use rspec

### PR DESCRIPTION
**Pull request by richardTowers**
What
----

Currently the "tests" only use the standard library. This is fine, but
it might be a bit more familiar to use the rspec library, since we use
that everywhere else on GOV.UK.

Why
---

This is partly motivated by #203 (which it will conflict with), in which
I ended up re-implementing bits of rspec's output. Partly I'm just bored.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -8,4 +8,19 @@ jobs:
...
+    - uses: actions/cache@v1
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gems-
+
+    - name: Bundle install
+      run: |
+        gem install bundler
```
Comment (issyl0) [inline .github/workflows/tests.yml:20]: You shouldn't need this, Ruby 2.6.6 bundles bundler with it.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -8,4 +8,19 @@ jobs:
     - uses: actions/setup-ruby@v1
       with:
         ruby-version: '2.6'
-    - run: ruby tests/well-formed-yaml.rb
+    - uses: actions/cache@v1
```
Comment (issyl0) [inline .github/workflows/tests.yml:11]: Do we need to bother with gem caching? There's only one gem.

Comment (richardTowers) [reply .github/workflows/tests.yml:11]: For now, but also _speeeed_

File: .github/workflows/tests.yml (hunk 1 above)
Comment (richardTowers) [reply .github/workflows/tests.yml]:
```
/opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/rubygems.rb:283:in `find_spec_for_exe': Could not find 'bundler' (2.1.4) required by your /home/runner/work/govuk-coronavirus-content/govuk-coronavirus-content/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:2.1.4`
	from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/rubygems.rb:302:in `activate_bin_path'
	from /opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle:23:in `<main>'
##[error]Process completed with exit code 1.
```

😭 y tho?

Comment (richardTowers) [reply .github/workflows/tests.yml:20]: So I removed it and the tests failed as above, so I've put it back.

**Comment by sihugh** [comment]
I'll swap the action to be called ci.yml etc _soon_

**Pull request merged**